File: PO_java/program/Organisms/Plants/Dandelion.java
package projekt.program.Organisms.Plants;

import projekt.program.Organism;
import projekt.program.Organisms.Plant;
import projekt.program.World;

public class Dandelion extends Plant {
    private static final int DANDELION_STRENGTH = 0;
    private static final int DANDELION_SEEDING_PROBABILITY = 5; // Example probability

    public Dandelion(World world, int x, int y) {
        super(world, x, y, DANDELION_STRENGTH, '.', DANDELION_SEEDING_PROBABILITY);
        // System.out.println("Creating Dandelion");
    }

    @Override
    public String getName() {
        return "Dandelion";
    }

    @Override
    public Organism createChild(int x, int y) {
        return new Dandelion(this.world, x, y);
    }

    @Override
    public void action(char command) {
        tryToSeed();
        tryToSeed();
        tryToSeed();
    }
}

File: PO_py/program/Organisms/Plants/Dandelion.py
from program.Organisms.Plant import Plant


class Dandelion(Plant):

    DANDELION_STRENGTH = 0
    DANDELION_SEEDING_PROBABILITY = 5

    def __init__(self, world, x, y):
        super().__init__(world, x, y, Dandelion.DANDELION_STRENGTH, '.', Dandelion.DANDELION_SEEDING_PROBABILITY)

    def get_name(self):
        return "Dandelion"

    def create_child(self, x, y):
        return Dandelion(self.world, x, y)

    def action(self, command):
        self.try_to_seed()
        self.try_to_seed()
        self.try_to_seed()
